gpii/node_modules/flowManager/src/UserSave.js

```diff
@@ -27,24 +28,23 @@ fluid.defaults("gpii.flowManager.userSave.handler", {
         handleRequest: {
             funcName: "gpii.flowManager.userSave.handleRequest",
             args: [
-                "{flowManager}.preferencesDataSource",
-                "{request}.req.params.userToken",
+                "{flowManager}.preferencesServer",
+                "{request}.req.params.gpiiKey",
                 "{request}.req.body",
                 "{request}"
             ]
         }
     }
 });
 
-gpii.flowManager.userSave.handleRequest = function (preferencesDataSource, userToken, preferences, request) {
-    // TODO: Slight faultiness here - we must encode a "missing value" as the empty string, since this value
-    // is transmitted directly as a URL request parameter to the upstream DataSource. We need to make sure that
-    // round-tripping via a DataSource's directModel (via "termMap" etc.) is not lossy
-    userToken = userToken || "";
-    var directModel = {
-        userToken: userToken
-    };
+gpii.flowManager.userSave.handleRequest = function (preferencesServer, gpiiKey, preferences, request) {
+    var promiseTogo;
     var writeMethod = request.req.method.toUpperCase();
-    var promise = preferencesDataSource.set(directModel, preferences, { writeMethod: writeMethod });
-    promise.then(request.events.onSuccess.fire, request.events.onError.fire);
+
+    if (writeMethod === "POST") {
+        promiseTogo = preferencesServer.createPreferences(preferences, null);
+    } else {
```

Comment:
I think this would be better if it was explicit:

```
else if (writeMethod === "PUT") {
    ...
} else {
    // Fire an error
}
```

Reply:
I will add the `else if` block.

But I think the `else` block is unnecessary because with [the control of requestHandlers](https://github.com/cindyli/universal/blob/GPII-2630/gpii/node_modules/flowManager/src/FlowManager.js#L136-L147) defined in `gpii.flowManager.save`, requests that don't match defined handlers will be rejected at the higher level in kettle and would never reach the `else`.